File: app/src/main/java/com/example/registerloginapp/ResetPasswordFragment.java
package com.example.registerloginapp;

import android.annotation.SuppressLint;
import android.content.pm.ActivityInfo;
import android.content.res.ColorStateList;
import android.content.res.XmlResourceParser;
import android.os.Bundle;

import androidx.fragment.app.Fragment;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import java.util.Objects;
import java.util.regex.Matcher;
import java.util.regex.Pattern;


public class ResetPasswordFragment extends Fragment implements View.OnClickListener {

    private  View view;
    private  EditText email, password,confirmPassword;
    private  Button loginButton;
    private  TextView  anyAccount;

    public ResetPasswordFragment() {
        // Required empty public constructor
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {

        view = inflater.inflate(R.layout.fragment_resetpassword, container, false);
        initViews();
        setListeners();
        return view;
    }

    private void initViews() {

        email = view.findViewById(R.id.confirm_email);
        password = view.findViewById(R.id.confirm_password);
        loginButton = view.findViewById(R.id.confirm_button);
        confirmPassword = view.findViewById(R.id.confirm_reset_password);
        anyAccount = view.findViewById(R.id.confirm_any_account);

        @SuppressLint("ResourceType") XmlResourceParser xrp = getResources().getXml(R.drawable.text_selector);
        try {
            ColorStateList csl = ColorStateList.createFromXml(getResources(),
                    xrp);
            anyAccount.setTextColor(csl);
        } catch (Exception ignored) {
        }
    }

    private void setListeners() {
        loginButton.setOnClickListener(this);
        anyAccount.setOnClickListener(this);
    }

    @Override
    public void onResume() {
        Objects.requireNonNull(getActivity()).setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_PORTRAIT);
        super.onResume();
    }

    @Override
    public void onClick(View v) {
        switch (v.getId()) {
            case R.id.confirm_button:
                if (checkValidation()) new MainActivity().replaceLoginFragment();
                break;

            case R.id.confirm_any_account:
                new MainActivity().replaceRegisterFragment();
                break;
        }
    }

    private boolean checkValidation() {
        boolean valid = true;
        String getConfirmPassword = confirmPassword.getText().toString();
        String getEmailId = email.getText().toString();
        String getPassword = password.getText().toString();

        Pattern p = Pattern.compile(Utils.regEx);
        Matcher m = p.matcher(getEmailId);

        if (    getEmailId.equals("") || getEmailId.length() == 0
                || getPassword.equals("") || getPassword.length() == 0
                || getConfirmPassword.equals("")
                || getConfirmPassword.length() == 0) {

            new CustomToast().Show_Toast(Objects.requireNonNull(getActivity()), view,
                    "All fields are required");
        valid = false; }
        else if (!m.find()) {
            new CustomToast().Show_Toast(Objects.requireNonNull(getActivity()), view,
                    "Email is Incorrect");
            valid = false;
        }
        else if (getConfirmPassword.length() < 6) {
            new CustomToast().Show_Toast(Objects.requireNonNull(getActivity()), view,
                    "Password must be at least 6 symbols");
            valid = false;
        }
        else if (!getConfirmPassword.equals(getPassword)) {
            new CustomToast().Show_Toast(Objects.requireNonNull(getActivity()), view,
                    "Both password doesn't match");
        valid = false; }
        else
            Toast.makeText(getActivity(), "Please Activate Your Account", Toast.LENGTH_SHORT)
                    .show();
        return valid;
    }
}

File: app/src/main/java/com/example/registerloginapp/ResetPasswordFragment.kt
package com.example.registerloginapp

import android.annotation.SuppressLint
import android.content.pm.ActivityInfo
import android.content.res.ColorStateList
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import android.widget.Toast
import androidx.fragment.app.Fragment
import kotlinx.android.synthetic.main.fragment_resetpassword.*
import java.util.*
import java.util.regex.Pattern

class ResetPasswordFragment : Fragment(), View.OnClickListener {
    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        v = inflater.inflate(R.layout.fragment_resetpassword, container, false)
        initViews()
        setListeners()
        return v
    }

    private fun initViews() {
        loginButton = v!!.findViewById(R.id.confirm_button)
        anyAccount = v!!.findViewById(R.id.confirm_any_account)
        @SuppressLint("ResourceType") val xrp = resources.getXml(R.drawable.text_selector)
        try {
            val csl = ColorStateList.createFromXml(resources,
                    xrp)
            anyAccount!!.setTextColor(csl)
        } catch (ignored: Exception) {
        }
    }

    private fun setListeners() {
        loginButton!!.setOnClickListener(this)
        anyAccount!!.setOnClickListener(this)
    }

    override fun onResume() {
        Objects.requireNonNull(activity)!!.requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_PORTRAIT
        super.onResume()
    }

    override fun onClick(vr: View) {
        when (vr.id) {
            R.id.confirm_button -> if (checkValidation()) MainActivity().replaceLoginFragment()
            R.id.confirm_any_account -> MainActivity().replaceRegisterFragment()
        }
    }

    private fun checkValidation(): Boolean {
        var valid = true
        val getConfirmPassword = confirm_reset_password.text.toString()
        val getEmailId =  confirm_email.text.toString()
        val getPassword = confirm_password.text.toString()
        val p = Pattern.compile(Utils.regEx)
        val m = p.matcher(getEmailId)
        if (getEmailId == "" || getEmailId.length == 0 || getPassword == "" || getPassword.length == 0 || getConfirmPassword == "" || getConfirmPassword.length == 0) {
            CustomToast().showtoast(Objects.requireNonNull(activity)!!, v!!, R.string.error_field_required)
            valid = false
        } else if (!m.find()) {
            CustomToast().showtoast(Objects.requireNonNull(activity)!!, v!!, R.string.error_email)
            valid = false
        } else if (getConfirmPassword.length < 6) {
            CustomToast().showtoast(Objects.requireNonNull(activity)!!, v!!, R.string.error_password)
            valid = false
        } else if (getConfirmPassword != getPassword) {
            CustomToast().showtoast(Objects.requireNonNull(activity)!!, v!!, R.string.error_reset)
            valid = false
        } else Toast.makeText(activity, R.string.reset_text, Toast.LENGTH_SHORT)
                .show()
        return valid
    }
    companion object {
        private var v: View? = null
        private var loginButton: Button? = null
        private var anyAccount: TextView? = null
    }
}